stl/inc/functional

```diff
@@ -1768,31 +1768,57 @@ private:
 
 // FUNCTION TEMPLATE _Build_boyer_moore_delta_2_table
 template <class _RanItPat, class _Pred_eq>
-void _Build_boyer_moore_delta_2_table(_Iter_diff_t<_RanItPat>* _Shifts, const _RanItPat _Pat_first,
-    const _Iter_diff_t<_RanItPat> _Pat_size, _Pred_eq& _Eq) {
-    // builds Boyer-Moore's delta_2 table from a pattern [_Pat_first, _Pat_first + _Pat_size)
-    // pre: _Shifts is a pointer to _Pat_size _Iter_diff_t<_RanIt>s
-    if (_Pat_size == 0) {
+void _Build_boyer_moore_delta_2_table(_Iter_diff_t<_RanItPat>* const _Shifts, const _RanItPat _Pat_first,
+    const _Iter_diff_t<_RanItPat> _Mx, _Pred_eq& _Eq) {
+    // Builds Boyer-Moore's delta_2 table from a pattern [_Pat_first, _Pat_first + _Mx).
+    // pre: _Shifts is a pointer to _Mx _Iter_diff_t<_RanItPat>s.
+    // This is the `dd'` algorithm from "Fast Pattern Matching In Strings" by Knuth, Morris, and Pratt (1977).
+    // Note that the published algorithm used 1-based indexing!
+    // This contains the "Rytter correction" from "Algorithms For String Searching: A Survey" by Baeza-Yates (1989),

```

Comment:
In the msvc version of this PR please check in a copy of `Baeza-Yates` to doc/ if you can. (I know we can't on GitHub due to licensing)

Reply:
Already done. (I'm grateful to Baeza-Yates, whose personal website linked to Google Scholar, which linked to https://dl.acm.org/doi/pdf/10.1145/74697.74700 containing the full PDF of his paper, with an exceedingly helpful summary of the history here.)

Reply:
I am unsure about links in comments but linking that paper might help.